src/components/tools/createAnnotationToolComponent.ts

```diff
@@ -0,0 +1,131 @@
+import { LPSAxisDir } from '@/src/types/lps';
+import vtkWidgetManager from '@kitware/vtk.js/Widgets/Core/WidgetManager';
+import { computed, defineComponent, h, toRefs } from 'vue';
+import type { Component, ComputedRef, PropType } from 'vue';
+import AnnotationContextMenu from '@/src/components/tools/AnnotationContextMenu.vue';
+import { useToolStore as useToolMetaStore } from '@/src/store/tools';
+import { AnnotationToolStore } from '@/src/store/tools/useAnnotationTool';
+import { Tools } from '@/src/store/tools/types';
+import { useCurrentImage } from '@/src/composables/useCurrentImage';
+import { getLPSAxisFromDir } from '@/src/utils/lps';
+import { useCurrentFrameOfReference } from '@/src/composables/useCurrentFrameOfReference';
+import {
+  useContextMenu,
+  useCurrentTools,
+} from '@/src/composables/annotationTool';
+import { Maybe } from '@/src/types';
+import { ContextMenuEvent } from '@/src/types/annotation-tool';
+
+export interface AnnotationToolComponentMeta<
+  ToolID extends string,
+  ToolStore extends AnnotationToolStore<ToolID>
+> {
+  type: Tools;
+  name: string;
+  useToolStore: () => ToolStore;
+  Widget2DComponent: Component;
+  PlacingWidget2DComponent: Component;
+}
+
+export function createAnnotationToolComponent<
```

Comment:
Vue component factories??  Cool! (like React =) For my edifcation, why a factory instead of passing everything as props to some generic compoenent?  

The other naive approach would have been to just start shoving all the logic into Composables.  Probably would have ended up with one big `useWidget2D` and RulerWidget2D/PolygonWidget2D would be shells.  Opprotunity for overrides/special-casing per tool.